### ISSUE TITLE
New model from same config not equal

### ISSUE THREAD

**Issue by ahmadmustafaanis** [tensorflow/tensorflow#40981]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.2.0
- Python version: 3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
```
 nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Fri_Feb__8_19:08:26_Pacific_Standard_Time_2019
Cuda compilation tools, release 10.1, V10.1.105
```
- GPU model and memory: Nvidia MX110 2GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"` v2.2.0-rc4-8-g2b96f3662b 2.2.0

**Describe the current behavior**
I created a basic model and saved its configuration using `get_config()`
**Describe the expected behavior**
I created a new model using the same configuration `tf.keras.Sequential.from_config()` and compared configurations of both, but both were not equal.
**Standalone code to reproduce the issue**

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
```
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
import json
import numpy as np

# Build the model

model = Sequential([
    Dense(units=32, input_shape=(32, 32, 3), activation='relu', name='dense_1'),
    Dense(units=10, activation='softmax', name='dense_2')
])
config_dict = model.get_config()

model_same_config = tf.keras.Sequential.from_config(config_dict)
print('Same config:', 
      model.get_config() == model_same_config.get_config())
print('Same value for first weight matrix:', 
      np.allclose(model.weights[0].numpy(), model_same_config.weights[0].numpy()))
```
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
@ahmadmustafaanis 

I have tried in colab with TF version 2.2.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/199604647d9093c66c7474be211320e7/untitled63.ipynb).You are also seeing the same behavior?.Thanks!

**Comment by ahmadmustafaanis** [comment]
Yes, I am seeing same behavior. Instead, It should be
```
Same config: True
Same value for first weight matrix: False
```
Because we are making it from the same configuration

**Comment by ravikyram** [comment]
@ahmadmustafaanis 

Can you please try with TF nightly versions and i am not seeing any issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/c49c67471fd67926420796e11789b98d/untitled63.ipynb).Please verify once and close the issue. Thanks!

**Comment by ahmadmustafaanis** [comment]
Yea, It is working in tf nightly. Can you fix it in tf2.2 too?

**Comment by ravikyram** [comment]
@ahmadmustafaanis 

It is fixed in TF-2.3-rc0 as well.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/ad4a2cfdd3201cd5402b6aa32012f794/untitled72.ipynb).You can use TF 2.3.Please,close this issue as the issue was fixed with TF 2.3 release.Thanks!
